src/py/flwr/superexec/deployment.py

```diff
@@ -84,12 +86,12 @@ def set_config(
         if not config:
             return
         if superlink_address := config.get("superlink"):
-            self.superlink = superlink_address
+            self.superlink = str(superlink_address)
```

Comment:
Before casting, we should check if the type is actually `str` and show a meaningful error message it it's not